### PR TITLE
cleanup wallet restore/create errors

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

File: src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx [hunk 1]
```diff
@@ -48,22 +45,18 @@ export const WalletSelectionScreen = () => {
...
-        await showErrorDialog(errorMessages.walletStateInvalid, intl)
-      } else if (error instanceof KeysAreInvalid) {
-        await showErrorDialog(errorMessages.walletKeysInvalidated, intl)
-        logout()
-      } else if (error instanceof NetworkError) {
-        await showErrorDialog(errorMessages.networkError, intl)
-      } else {
-        await showErrorDialog(errorMessages.generalError, intl, {message: error.message})
-      }
+      InteractionManager.runAfterInteractions(() => {
```
Comment (wolverineks) [inline src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx:50]: closes loading modal, then display native error modal

File: src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx [hunk 2]
```diff
@@ -48,22 +45,18 @@ export const WalletSelectionScreen = () => {
...
+        return error instanceof SystemAuthDisabled
```
Comment (wolverineks) [inline src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx]: KeysAreInvalid is deprecated

File: src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx [hunk 3]
```diff
@@ -48,22 +45,18 @@ export const WalletSelectionScreen = () => {
...
+          ? showErrorDialog(errorMessages.enableSystemAuthFirst, intl)
```
Comment (wolverineks) [inline src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx]: dont await in `onError` unless you want to resolve before `onSettled`

File: src/hooks/index.ts [hunk 4]
```diff
@@ -84,6 +92,7 @@ export const useCrashReports = () => {
 
 // WALLET
 export const useWallet = (wallet: YoroiWallet, event: WalletEvent['type']) => {
+  const walletManager = useWalletManager()
```
Comment (wolverineks) [inline src/hooks/index.ts:95]: dependency injection for testing

File: src/yoroi-wallets/walletManager.ts [hunk 5]
```diff
@@ -26,7 +26,6 @@ import {WALLET_IMPLEMENTATION_REGISTRY} from './types/other'
 
 export class WalletClosed extends ExtendableError {}
 export class SystemAuthDisabled extends ExtendableError {}
-export class KeysAreInvalid extends ExtendableError {}
```
Comment (wolverineks) [inline src/yoroi-wallets/walletManager.ts:29]: deprecated

File: src/SelectedWallet/WalletSelection/WalletSelectionScreen.stories.tsx [hunk 6]
```diff
@@ -104,60 +104,8 @@ storiesOf('WalletSelectionScreen', module)
...
-    id: Math.random(),
-    name: String(Math.random()),
-  },
-  {
-    ...mocks.walletMeta,
-    id: Math.random(),
-    name: String(Math.random()),
-  },
-]
+const mockWalletMetas = new Array(15).fill(null).map(() => ({
```
Comment (stackchain) [inline src/SelectedWallet/WalletSelection/WalletSelectionScreen.stories.tsx:107]: simpler mocks

File: src/SelectedWallet/WalletSelection/WalletSelectionScreen.stories.tsx [hunk 7]
```diff
@@ -67,23 +67,6 @@ storiesOf('WalletSelectionScreen', module)
       <WalletSelectionScreen />
     </WalletManagerProvider>
   ))
-  .add('error, system auth disabled ', () => (
```
Comment (stackchain) [inline src/SelectedWallet/WalletSelection/WalletSelectionScreen.stories.tsx]: depreacted

File: src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx [hunk 8]
```diff
@@ -49,9 +48,7 @@ export const WalletSelectionScreen = () => {
       closeWallet()
 
       InteractionManager.runAfterInteractions(() => {
-        return error instanceof SystemAuthDisabled
```
Comment (stackchain) [inline src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx]: deprecated

File: src/yoroi-wallets/walletManager.ts [hunk 9]
```diff
@@ -25,7 +25,6 @@ import {
 import {WALLET_IMPLEMENTATION_REGISTRY} from './types/other'
 
 export class WalletClosed extends ExtendableError {}
-export class SystemAuthDisabled extends ExtendableError {}
```
Comment (stackchain) [inline src/yoroi-wallets/walletManager.ts:28]: deprecated

File: src/WalletInit/SaveNanoX/SaveNanoXScreen.tsx [hunk 10]
```diff
@@ -13,10 +17,19 @@ export const SaveNanoXScreen = () => {
...
   const {createWallet, isLoading} = useCreateBip44Wallet({
     onSuccess: () => {
       resetToWalletSelection()
     },
+    onError: (error) => {
+      InteractionManager.runAfterInteractions(() => {
+        const newLocal = error instanceof NetworkError
+        return newLocal
+          ? showErrorDialog(errorMessages.networkError, intl)
+          : showErrorDialog(errorMessages.generalError, intl, {message: error.message})
```
Comment (stackchain) [inline src/WalletInit/SaveNanoX/SaveNanoXScreen.tsx]: Do we need `newLocal`?
```suggestion
        return error instanceof NetworkError
          ? showErrorDialog(errorMessages.networkError, intl)
          : showErrorDialog(errorMessages.generalError, intl, {message: error.message})
```

Comment (wolverineks) [reply src/WalletInit/SaveNanoX/SaveNanoXScreen.tsx]: vscode autorefactor...

Comment (wolverineks) [reply src/WalletInit/SaveNanoX/SaveNanoXScreen.tsx]: so...no

Comment (stackchain) [reply src/WalletInit/SaveNanoX/SaveNanoXScreen.tsx]: it doesn't make sense to me, running on my VSCode the result is even different: `
      const newLocal = error instanceof NetworkError
        ? showErrorDialog(errorMessages.networkError, intl)
        : showErrorDialog(errorMessages.generalError, intl, { message: error.message })
      InteractionManager.runAfterInteractions(() =>
        newLocal
      )
`

**Review by stackchain — APPROVED** [review]

**Review by banklesss — APPROVED** [review]

**Pull request merged**